### PR TITLE
introduce olm catalog based variant of e2e test

### PR DESCRIPTION
**Pull request by bparees**
This pr adds a workflow that will
1) build a new operator-registry image that can act as a catalogsource serving the elasticsearch operator based on manifest content from the PR/repo itself, referencing operator/operand images based on env vars(so they can potentially come from what CI has built, or from what you have built+pushed locally and want to test)

2) deploy that operator-registry image and create a catalogsource for it

3) install the operator from that catalogsource

This means testing done based on this workflow will test the same CSV/manifest content that the final published operator will use, and it will test that CSV/manifest content by deploying it through the normal OLM operator install workflow, instead of instantiating the manifest content manually.

once this merges and we add a test job to openshift/release that leverages it, i'll go back and remove the old e2e test files.

### CODE REVIEW

**Comment by bparees** [comment]
/assign @jcantrill 

i'm happy to give a walkthrough of this PR to the team at some point.  we're going to have to do something similar in CLO(but probably more complicated, because the CLO test jobs will need to deploy both the EO catalog and the CLO catalog).

File: Makefile [hunk 1]
```diff
@@ -154,3 +154,43 @@ scale-olm:
...
 	hack/undeploy.sh
 .PHONY: undeploy
+
+
+# to use these targets, ensure the following env vars are set:
+# either each IMAGE env var:
+# IMAGE_ELASTICSEARCH_OPERATOR_REGISTRY
+# IMAGE_ELASTICSEARCH_OPERATOR
+# IMAGE_ELASTICSEARCH6
+# IMAGE_LOGGING_KIBANA
```
Comment (ewolinetz) [inline Makefile]: are the naming conventions different due to the underlying scripts?

Comment (bparees) [reply Makefile]: i was trying to use the names that api.ci would inject into the test jobs(if they were so configured) what naming conventions are you referring ti?

Comment (bparees) [reply Makefile]: i guess this should be logging_kibana6

File: olm_deploy/scripts/catalog-deploy.sh [hunk 2]
```diff
@@ -1,10 +1,10 @@
 #!/bin/sh 
 set -eou pipefail
 
-export IMAGE_ELASTICSEARCH_OPERATOR=${IMAGE_ELASTICSEARCH_OPERATOR:-quay.io/openshift/origin-elasticsearch-operator:latest}
-export IMAGE_ELASTICSEARCH6=${IMAGE_ELASTICSEARCH6:-quay.io/openshift/origin-logging-elasticsearch6:latest}
-export IMAGE_LOGGING_KIBANA6=${IMAGE_LOGGING_KIBANA6:-quay.io/openshift/origin-logging-kibana6:latest}
-export IMAGE_OAUTH_PROXY=${IMAGE_OAUTH_PROXY:-quay.io/openshift/origin-elasticsearch-proxy:latest}
+export IMAGE_ELASTICSEARCH_OPERATOR=${IMAGE_ELASTICSEARCH_OPERATOR:-registry.svc.ci.openshift.org/ocp/4.5:elasticsearch-operator}
```
Comment (ewolinetz) [inline olm_deploy/scripts/catalog-deploy.sh]: is it possible to use an env var for the image name version? (`4.5`) to make it easier for updating in our future releases?

Comment (bparees) [reply olm_deploy/scripts/catalog-deploy.sh]: in general i expect this var to be populated by the ci job, or explicitly set by the developer, but yeah i can make the version an env var for the defaulting case.

Comment (bparees) [reply olm_deploy/scripts/catalog-deploy.sh]: updated

**Comment by ewolinetz** [comment]
/approve

> i'm happy to give a walkthrough of this PR to the team at some point. we're going to have to do something similar in CLO(but probably more complicated, because the CLO test jobs will need to deploy both the EO catalog and the CLO catalog).

would it make sense to eventually have this logic in another repo that different OLM operators can then use?

**Comment by bparees** [comment]
> would it make sense to eventually have this logic in another repo that different OLM operators can then use?

i guess we can see how much of it can be factored out and reused vs how much is tied to the specific operator (e.g. the list of image names that need to be substituted)

**Pull request merged**